optimum/onnxruntime/modeling_seq2seq.py

```diff
@@ -0,0 +1,385 @@
+            config=onnx_config_decoder,
+            opset=onnx_opset,
+            output=save_dir.joinpath(ONNX_DECODER_NAME),
+        )
+
+        # Export the decoder with the past key values
+        export(
+            preprocessor=tokenizer,
+            model=decoder_with_lm_head,
+            config=onnx_config_decoder_with_past,
+            opset=onnx_opset,
+            output=save_dir.joinpath(ONNX_DECODER_WITH_PAST_NAME),
+        )
+
+        kwargs["config"] = model.config.__dict__
+        return cls._from_pretrained(save_dir.as_posix(), **kwargs)
+
+    def generate(self, *args, **kwargs):
+        return self.model.generate(*args, **kwargs)
+
+    def forward(self, *args, **kwargs):
+        return self.model.forward(*args, **kwargs)
+
+
+class ORTEncoder(torch.nn.Module):
+    def __init__(self, encoder: onnxruntime.InferenceSession):
+        super().__init__()
+        self.encoder = encoder
+        self.main_input_name = "input_ids"
+        self.output_names = {output_key.name: idx for idx, output_key in enumerate(self.encoder.get_outputs())}
+
+    def forward(
+        self,
+        input_ids: torch.LongTensor = None,
+        attention_mask: Optional[torch.Tensor] = None,
+        **kwargs,
+    ) -> BaseModelOutput:
+
+        onnx_inputs = {
+            "input_ids": input_ids.cpu().detach().numpy(),
+            "attention_mask": attention_mask.cpu().detach().numpy(),
+        }
+
+        # Run inference
+        outputs = self.encoder.run(None, onnx_inputs)
+
+        return BaseModelOutput(last_hidden_state=torch.from_numpy(outputs[self.output_names["last_hidden_state"]]))
+
+
+class ORTDecoder(torch.nn.Module):
+    def __init__(self, decoder: onnxruntime.InferenceSession):
+        super().__init__()
+        self.decoder = decoder
+        self.input_names = {input_key.name: idx for idx, input_key in enumerate(self.decoder.get_inputs())}
+        self.output_names = {output_key.name: idx for idx, output_key in enumerate(self.decoder.get_outputs())}
+
+    def forward(
+        self, input_ids, encoder_hidden_states, encoder_attention_mask, past_key_values=None
+    ) -> Seq2SeqLMOutput:
+
+        onnx_inputs = {
+            "input_ids": input_ids.cpu().detach().numpy(),
+            "encoder_attention_mask": encoder_attention_mask.cpu().detach().numpy(),
+        }
+
+        # Add the encoder_hidden_states inputs when needed
+        if "encoder_hidden_states" in self.input_names:
+            onnx_inputs["encoder_hidden_states"] = encoder_hidden_states.cpu().detach().numpy()
+
+        if past_key_values is not None:
+            # Flatten the past_key_values
+            past_key_values = [mha for past_key_value in past_key_values for mha in past_key_value]
+
+            # Add the past_key_values to the decoder inputs
+            for i, past_key_value in enumerate(past_key_values):
+                onnx_inputs[f"past_key_values_{i}.1"] = past_key_value.cpu().detach().numpy()
```

Comment:
I would assume everything is `np.ndarray` here and remove any dependency on PyTorch at this stage.